gprofiler/utils/fs.py

```diff
@@ -76,6 +76,43 @@ def is_owned_by_root(path: Path) -> bool:
     return statbuf.st_uid == 0 and statbuf.st_gid == 0
 
 
+def is_owned_by_user(path: Path) -> bool:
+    statbuf = path.stat()
+    return statbuf.st_uid == os.geteuid() and statbuf.st_gid == os.getegid()
+
+
+def mkdir_owned_user(path: Union[str, Path], mode: int = 0o755) -> None:
```

Comment:
This behavior is not needed when we're not root. So - either add a wrapper, which checks if we're root and calls it OR simply `mkdir`s, or if there's 1-2 calls sites, add such `if` directly:
```
if is_root():
   mkdir_owned_root(...)
else:
   mkdir(...)
```